PolyEngine/Engine/Src/ECS/World.hpp

```diff
@@ -121,16 +121,61 @@ namespace Poly {
 				return **this;
 			}
 
-			ComponentIterator& operator++() { ++primary_iter; return *this; }
-			ComponentIterator operator++(int) { ComponentIterator ret(primary_iter); ++primary_iter; return ret; }
-			ComponentIterator& operator--() { --primary_iter; return *this; }
-			ComponentIterator operator--(int) { ComponentIterator ret(primary_iter); --primary_iter; return ret; }
+			ComponentIterator& operator++() { Increment(); return *this; }
+			ComponentIterator operator++(int) { ComponentIterator ret(primary_iter); Increment(); return ret; }
+			ComponentIterator& operator--() { Decrement(); return *this; }
+			ComponentIterator operator--(int) { ComponentIterator ret(primary_iter); Decrement(); return ret; }
+
+		private:
+			template<int zero = 0>
+			bool HasComponents(const Entity* entity) const
+			{
+				return true;
+			}
+			template<typename Component, typename... Rest>
+			bool HasComponents(const Entity* entity) const
+			{
+				if (entity->template GetComponent<Component>())
```

Comment:
Change this whole block to 
`return entity->template GetComponent<Component>() && HasComponents<Rest...>(entity);`

Reply:
If possible also add to entity method `HasComponent<>()` which calls `GetComponent<>()` underneath for now, but returns boolean.